### PR TITLE
Create the oauth2 settings and configuration in registrar

### PR DESCRIPTION
**Pull request by schenedx**
## Description

With this change, LMS users can authenticate with registrar API through DOT
See https://openedx.atlassian.net/browse/EDUCATOR-4147

### CODE REVIEW

**Comment by schenedx** [comment]
This is related to https://github.com/edx/configuration/pull/5045

**Comment by schenedx** [comment]
@edx/masters-neem Please review

**Review by iloveagent57 — APPROVED** [review]

File: registrar/apps/api/v1/views.py [hunk 1]
```diff
@@ -33,6 +34,7 @@ class ProgramReadOnlyViewSet(viewsets.ReadOnlyModelViewSet):
         '404':
             description: Program key not found.
     """
+    authentication_classes = (JwtAuthentication,)
```
Comment (iloveagent57) [inline registrar/apps/api/v1/views.py:37]: This is going to break all of the existing tests.

**Review by iloveagent57 — COMMENTED** [review]
Ignore my previous approval...

File: registrar/apps/api/v1/views.py (hunk 1 above)
Comment (schenedx) [reply registrar/apps/api/v1/views.py:37]: It's fixed now.

File: registrar/apps/api/v1/tests/test_views.py [hunk 2]
```diff
@@ -60,10 +61,11 @@ def setUp(self):
     def login(self, user):
         return self.client.login(username=user.username, password=USER_PASSWORD)
 
-    def get(self, path):
+    def get(self, path, user, is_staff=False):
```
Comment (kdmccormick) [inline registrar/apps/api/v1/tests/test_views.py]: The `is_staff` parameter is unnecessary as it is a field on `user` (`user.is_staff`)

File: registrar/settings/test.py [hunk 3]
```diff
@@ -15,3 +15,12 @@
...
 # END IN-MEMORY TEST DATABASE
+
+OAUTH2_PROVIDER_URL = 'https://test-provider/oauth2'
+SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT = OAUTH2_PROVIDER_URL
+
+JWT_AUTH['JWT_ISSUERS'] = [{
+    'SECRET_KEY': SOCIAL_AUTH_EDX_OAUTH2_SECRET,
+    'ISSUER': OAUTH2_PROVIDER_URL,
+    'AUDIENCE': SOCIAL_AUTH_EDX_OAUTH2_KEY,
+}]
```
Comment (kdmccormick) [inline registrar/settings/test.py]: nit: Newline

**Review by iloveagent57 — APPROVED** [review]
Nice.  Where'd you pull that `JwtMixin` class from?

**Comment by schenedx** [comment]
@iloveagent57 How do you know I pulled that class from somewhere? I can't write that myself?
I "lifted" it from credentials

**Review by kdmccormick — COMMENTED** [review]
Code looks good, other than a couple of nits. Make sure when you rebase you put your edX email in the commit ;)

However, I'm having trouble getting it working on my local devstack. Registrar redirects me to LMS to log in, but when I log in as `honor@example.com`, it redirects me to the Registrar admin panel, authenticated as the `edx` user.

**Comment by kdmccormick** [comment]
@schenedx sorry, I didn't mean put your email in the commit message. I mean you're authoring the commit under a different email (looks like `schen@C02XW01JJK7L.tld`) so things like git-blame won't work properly. You want to run `git config --global user.email "schen@edx.org"` before re-committing.

**Comment by schenedx** [comment]
@kdmccormick Got it. Thank you for the instruction.

**Review by kdmccormick — APPROVED** [review]
LGTM! Please choose the Rebase & Merge option.

**Pull request merged**